### ISSUE TITLE
Windows IO subsystem drops out, leaving system running but deadlocked

### ISSUE THREAD

**Issue by smopucilowski** [maharmstone/btrfs#438]
Over the last year, using winbtrfs versions 1.7.3 through to current 1.7.9 on laptops with operating system on ntfs and data on btrfs, we have been experiencing soft hangs. We have observed this across multiple machines with varying hardware.

Specifically, Windows 10/11 operates normally with light btrfs use with subvolumes, until some point suddenly the entire IO storage stack stops processing requests. Programs running in memory, e.g. internet chat or music streaming, continue to function until they make their own storage IO request. They then soft hang waiting for their own IO requests to complete.
Since the IO subsystem is not responding, it's not possible to interact with anything requiring further storage IO, i.e. bringing up a task manager or explorer. Windows system logs don't report anything when the system hangs.
It's as if the entire IO subsystem is stuck in some deadlock. I would expect after 60 seconds for everything to crash out saying disk timeouts, however, IO stalls don't time out even after leaving the machines for hours.

The only solution is to hard reset to laptops.

Possibly linked to laptop use with varying sleep states?

How may I go about assisting debugging this? Is there a remote debug facility in windows that can dump stacks/crashes over the network?

**Comment by kwencel** [comment]
I can confirm this is happening exactly as you described. On a Windows 10 PC, so I don't think it is linked to laptops or sleep states.

Unfortunately, I don't have any logs or steps to reproduce. It just happens sometimes.

**Comment by ticpu** [comment]
This also happens quite often to me. Usually minutes after doing subvolume operations or writing to multiple subvolumes in quick successions. I just can't tell the exact trigger and have to force reset the machine. Windows 10 and is happening since I started using WinBtrfs. I just never created an issue since I don't know where to start debugging this.

**Comment by alucryd** [comment]
Started happening here recently on 2 different computers as well. One is a regular partition, the other is a striped partition across 2 SSDs. Sometimes it comes back on its own after a while, sometimes I have to hard reset. Seems to be triggered by heavy writes (usually happens while downloading games on steam for instance). It did not happen a couple versions ago so it looks like a new regression.

**Comment by PJB3005** [comment]
Can also confirm having similar problems all the time, no idea how to even start to debug this.

**Comment by maharmstone** [comment]
Whatever it is, it's not something obvious, and quite possibly something introduced with later versions of Windows. I'm currently working on a comprehensive test suite which will hopefully help in diagnosing this.

**Comment by mhetzi** [comment]
This problem can be reliably reproduced, by playing forza horizon 5. Can i do something to help debug this?

**Comment by PJB3005** [comment]
In my experience running steam games off my BTRFS partition, it seems *really* consistent once it starts happening for my games. Like, Steam consistently locking up while updating, or the game consistently locking up while starting. So I can also reliably reproduce this if I have some debugging instructions.

I tried using the kernel event logging thing mentioned in the readme but couldn't get it to show anything.

**Comment by sebres** [comment]
FWIW, it seems like I got it also deadlocked with latest WinBtrfs (v1.7.9, with enabled zstd compression) just using robocopy copying a large folder with /MIR option from NTFS drive to BTRFS drive.
Almost immediate a BTRFS drive was not accessible elsewhere, e. g. calling properties in file explorer for the drive showing the error "The process cannot access the file because it is being used by another process".
The robocoby is still running further a bit, but after several minutes it starts to stutter with "ERROR 32 (0x00000020) Accessing Destination Directory". 
At end it was completely locked, so not accessible in any process, until restarted.
I was able to reproduce the same behavior few times.

**Comment by jarrard** [comment]
I don't know if its related or not or if I need to open a new ticket but I've been noticing when I updated games (large games) on steam my btrfs partition will cease to mount or exist unless I select on it and refresh to bring it back to life.

Could this be a similar IO failure?   I AM using disk compression with zstd:3 so maybe that is related. 

So far just using the SSD/NVMe drives normally is fine, its just when steam does its ultra intensive IO operations to update/scan something; the partition shows no size value and appears to stop working as steam stalls (until I select drive in explorer and refresh it a few times).

**Comment by RedKage** [comment]
Same thing happened to me.
BTRFS on a 2TB disk, single partition
Steam library on it.
After downloading Killing Floor 2 and trying to install -> it froze
Game is about 50Gb

Since it froze, I couldn´t do anything. Steam crashed. Explorer was stuck, and restarting was stuck...
I had to power off while "Restarting ..." was printed.

Later did a "scrub" from windows and it found no error... but right after that got read-only, with cyclic redundancy error.
Then went to Linux, fstab crashed because it couldn't mount the drive no more.
Had to comment out my fstab in rescue mode.

The error:
```
Opening filesystem to check...
checksum verify failed on 1223022346240 wanted 0x00000000 found 0xb6bde3e4
checksum verify failed on 1223022346240 wanted 0x00000000 found 0xb6bde3e4
checksum verify failed on 1223022346240 wanted 0x00000000 found 0xb6bde3e4
bad tree block 1223022346240, bytenr mismatch, want=1223022346240, have=0
ERROR: cannot read chunk root
ERROR: cannot open file system
```

These errors are probably due to me power restarting.
The actual issue is the fact that it deadlocked. Not sure if I can help with providing more info since nothing made any obvious "crash" 

I'm in the process of doing a `chunk-recover`. I don´t care if my data is lost, I'm trying out WinBTRFS...
But wow, that's very unstable.

**Comment by jarrard** [comment]
I've been testing my BTRFS partitions under Linux for a while now without too many issues; but loosing power is a pretty big nasty thing to happen for solid state drives, most times results in some files being messed.  It does seem like the drives are a little more sensitive under windows then Linux. 

Fortunately I don't get too many of these IO errors and if I do typically clicking on the stalled btrfs drive in windows explorer eventually brings it back just fine and steam resumes its operations. 

There are more detailed logs in the windows event viewer which might be more useful for figuring out whats happening. I forget exactly where however. They produce .evtx files.

**Comment by Gummar** [comment]
> FWIW, it seems like I got it also deadlocked with latest WinBtrfs (v1.7.9, with enabled zstd compression) just using robocopy copying a large folder with /MIR option from NTFS drive to BTRFS drive. Almost immediate a BTRFS drive was not accessible elsewhere, e. g. calling properties in file explorer for the drive showing the error "The process cannot access the file because it is being used by another process". The robocoby is still running further a bit, but after several minutes it starts to stutter with "ERROR 32 (0x00000020) Accessing Destination Directory". At end it was completely locked, so not accessible in any process, until restarted. I was able to reproduce the same behavior few times.

The same thing happens here when using robocopy to move 70gb of files from a ntfs disk to a mounted btrfs VHD stored inside a ntfs disk, except robocopy runs infinitely without showing any error and doing no changes to any files, any process that tries to access the mounted partition freezes with no way to close them. I'm using v1.7.9 with zstd:6 and /csum:xxhash. I'm running Windows 7 x64.

**Comment by devurandom** [comment]
It happened on my computer again, when Forza Horizon 5 was running. Windows 10 was unable to close the game or steam and could not shut down either (it would just hang with a dark screen). Afterwards on Linux I ran `btrfsck`:

```
❯ sudo btrfsck [REDACTED]
Opening filesystem to check...
Checking filesystem on [REDACTED]
UUID: [REDACTED]
[1/7] checking root items
[2/7] checking extents
super bytes used 1686982955008 mismatches actual used 1683220516864
ERROR: errors found in extent allocation tree or chunk allocation
[3/7] checking free space tree
free space info recorded 35 extents, counted 34
there is no free space entry for 609915752448-609915764736
cache appears valid but isn't 608842022912
there is no free space entry for 622789128192-622789132288
there is no free space entry for 622789128192-622800666624
cache appears valid but isn't 621726924800
free space info recorded 1 extents, counted 0
there is no free space entry for 698413178880-698413187072
there is no free space entry for 698413178880-699036336128
cache appears valid but isn't 697962594304
[4/7] checking fs roots
[5/7] checking only csums items (without verifying data)
[6/7] checking root refs
[7/7] checking quota groups skipped (not enabled on this FS)
found 1683220516864 bytes used, error(s) found
total csum bytes: 1640095780
total tree bytes: 3762438144
total fs tree bytes: 1400471552
total extent tree bytes: 473677824
btree space waste bytes: 581650624
file data blocks allocated: 1710374117376
 referenced 1882496815104
```

**Comment by devurandom** [comment]
Possibly related: https://github.com/maharmstone/btrfs/issues/418

**Comment by AEAEAEAE4343** [comment]
I'm having the same issue on a 2TB striped Btrfs volume. Though the symptoms are the same, I don't think the entire IO subsystem is dropping out. 

The issue occurs while actively using the drive for me, so I let open a command prompt while the drive was in use. When my system started to freeze, I checked what was accessible within that command prompt. My system drive (NTFS) and a Btrfs drive I was not actively using were both accessible, and I could read and write to both drives.

It seems like there's just a file (or maybe disk) operation hanging on the Btrfs drive, taking most of the system down with it, due to stuff like explorer depending on that drive. This suspicion is further confirmed by Linux: every time this happens, another unrecoverable error gets added to the volume (`btrfs scrub`), probably indicating an unsuccesful write.

More information about the volume itself:
```
Device size:		1.8 TB
Device allocated:	614.0 GB
Device unallocated:	1.2 TB
Data ratio:		1.00
Metadata ratio:		2.00
Data, RAID0: size: 608.0 GB, used: 607.4 GB
Disk 1, partition 1	304.0 GB
Disk 2, partition 1	304.0 GB
Metadata, RAID1: size: 3.0 GB, used: 1.4 GB
Disk 1, partition 1	3.0 GB
Disk 2, partition 1	3.0 GB
System, RAID1: size: 8.0 MB, used: 64.0 KB
Disk 1, partition 1	8.0 MB
Disk 2, partition 1	8.0 MB
Unallocated:
Disk 1, partition 1	624.5 GB
Disk 2, partition 1	624.5 GB
```

**Comment by sebres** [comment]
> I'm having the same issue...
> and I could read and write.

Then you had probably another issue.
At least in my case the (robo)copy process with following "freeze" always causes total deadlock and btrfs drive is inaccessible completely, neither for write nor for read operations unless computer gets restarted.

**Comment by sebres** [comment]
> but my other partitions work fine.

Why they shouldn't? 
The work with kernel driver takes place asynchronously via [IoQueue and WorkItems](https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/system-worker-threads) and tasks are isolated from user space per single drive/volume.
If such queue or worker gets deadlocked it'd only affect single volume solely.

> explorer freezes regardless of what directory is selected

This is happening because of the matter how explorer renders (and reads) the items in tree in left part of window, for instance if it has some reference to directory from affected drive in "Quick access" or because it is still accessing the drive info or something similar. Anyway note that the explorer is normally a single process (no matter how many folder windows you'd open there), at least unless the checkbox "Launch folder windows in a separate process" gets checked in "Folder options / View".

**Comment by AEAEAEAE4343** [comment]
> Why they shouldn't?
I never said it should. It’s just that the original question suggests that:
```
suddenly the entire IO storage stack stops processing requests. Programs running in memory, e.g. internet chat or music streaming, continue to function until they make their own storage IO request. They then soft hang waiting for their own IO requests to complete.
Since the IO subsystem is not responding, it's not possible to interact with anything requiring further storage IO, i.e. bringing up a task manager or explorer.
```
All I’m saying is that this isn’t the case, only the drive that gets stuck is inaccessible, which makes sense. If the entire IO subsystem were to drop out, I think the system would bluescreen immediately, not partially freeze up. 
> for instance if it has some reference to directory from affected drive in "Quick access" or because it is still accessing the drive info or something similar
In my case, I have shortcuts to my Btrfs drive on my desktop which freeze explorer. For file explorer itself, it probably tries reading metadata.

**Comment by ArinL** [comment]
Is anyone still experiencing these issues on v1.8?

**Comment by jarrard** [comment]
I've been mostly using Linux so haven't had much time to test. But I also haven't seen any winbtrfs updates for a while.

**Comment by smopucilowski** [comment]
I had a btrfs partition corrupt itself after a recovery power cycle (no valid root node), so I'm a little shy to experiment further and am running with read-only mounting for now.

**Comment by AEAEAEAE4343** [comment]
After updating to 1.8, I can no longer boot my system at all.

**Comment by AEAEAEAE4343** [comment]
I think the problem is resolved. Speeds have improved and after testing by copying a ton of data (~100GB) I haven't had any issues.

**Comment by smopucilowski** [comment]
Hit the issue again on winbtrfs v1.8 on a freshly formatted btrfs partition in Linux (2tb, space_cache=v2,compress=zstd), copying a few gigabytes into a subvolume. Scrub reports no issues, no corrupted partition encountered yet.

**Comment by smopucilowski** [comment]
I've had relative success with infrequent large file copies (many gigabytes, queue depth of 1).

It feels like this bug triggers on multiple small read/writes, or IO with deep queue depth.

**Comment by gmmoreira** [comment]
I still have the problem on v1.8. In my case it usually happens when steam is downloading or updating a game. I have to hard reset my computer because it becomes unresponsive.

**Comment by Kamilcuk** [comment]
I have the same issue. Happens when unpacking a 11GB .zip file from and also onto a btrfs partition.

**Comment by Minusman666** [comment]
Hello , I also have a problem with my btrfs Nvme Raid10 federation. When I work with software that performs many read and write operations on the raid10, the disk array slows down and almost stops. To stop the process, I then have to perform the hard reset. 
I've been using WinBtrfs for a few years now and haven't had such problems under Windows 10. Somehow Windows 11 does not harmonize with WinBtrfs 1.8.1. I had also performed another downgrade to 1.7.8, but had brought nothing.
Is there a solution for the problem? I could not find anything in the topic.

With best regards.

**Comment by jarrard** [comment]
Was able to invoke this issue when having Mod Organizer 2 open and downloading a large mod into a mod folder.

x is not accessible, the process cannot access the file because it is being used by another process.  

In this case I'm just browsing with FME and downloading via firefox which I guess is the other process....

I thought winbtrfs had async io but its acting like single threaded I/O.......

**Comment by Kingtous** [comment]
I still have the problem on v1.8.1.

I'm copying a windows vm from a portable ssd which contains tons of development envrionment, about 250GB. After I right click the destination folder and click `paste` button, the whole win11 freezes. If I unplug my portable ssd, the system is still stucking. And then the system crashes and show bluescreen.

Is there any old version of `winbtrfs` working? It's not easy to transfer the vm because of its giant size and reformat my ssd. Hope it will fix it soon.

**Comment by mikejzx** [comment]
Also been having this issue on WinBtrfs v1.8.1 on Windows 8.1 (therefore I don't think it would be due to something newly-introduced in recent Windows versions).  It seems to occur when generally performing I/O tasks on the drive for some period of time, and is almost guaranteed to occur (at least in my case) when performing heavy I/O tasks (e.g. installing Unreal Engine, which caused multiple lockups for me).

**Comment by CSKnock** [comment]
have there been any new discoveries towards this?

**Comment by jarrard** [comment]
> have there been any new discoveries towards this?

I'm mostly under Linux now but do have 3 btrfs partitions that'll end up testing this driver sooner or later when I move back to Windows11 for testing stuff.  (I tend to use linux for 6months then 3months on windows, etc etc..)

**Comment by ticpu** [comment]
I can only affirm that after months, it is mostly happening when there are many concurrent operations running and is worse when adding Btrfs-specific operations like creating a snapshot.
Nothing I can easily reproduce, however, 2 copies in parallel or a copy and a Steam game update running, make a snapshot of that subvolume and you usually end up with a frozen FS that needs a hard reset.

**Comment by maharmstone** [comment]
> have there been any new discoveries towards this?

No, because I've not been able to reproduce it.

**Comment by alucryd** [comment]
Downloading any steam game is enough to trigger the bug almost every time here. Granted this is at fiber internet speeds, and on an array of 3 SATA SSDs. At the very least, it's possible it is exacerbated with btrfs spanning multiple disks.
I learned to download my games on Linux even when I want to play them on Windows (which is less and less these days with atrocious Windows releases that run better on Linux anyway), but it's definitely reproducible and makes the driver very much unusable under these circumstances. It's pretty much perfect outside of that though, so if that could be ironed out it would be awesome.

**Comment by maharmstone** [comment]
> Downloading any steam game is enough to trigger the bug almost every time here. Granted this is at fiber internet speeds, and on an array of 3 SATA SSDs. At the very least, it's possible it is exacerbated with btrfs spanning multiple disks. I learned to download my games on Linux even when I want to play them on Windows (which is less and less these days with atrocious Windows releases that run better on Linux anyway), but it's definitely reproducible and makes the driver very much unusable under these circumstances. It's pretty much perfect outside of that though, so if that could be ironed out it would be awesome.

Interesting. Does this still happen if you throttle Steam's download speeds to, say, 5 MB/s?

**Comment by jarrard** [comment]
If you play a game that has excessive memory usage it tends of have aggressive IO activity which can cause the partition to become unavailable/mountfail but I'd have to go back to windows and test this out again until point of failure to figure out which of my games can cause it. 

I intend to return to Windows soon to do some testing as I'd like this resolved if it indeed does still happen in general use cases. 

I also wonder if processing large video encoding would trigger it?

**Comment by lesderid** [comment]
> > Downloading any steam game is enough to trigger the bug almost every time here. Granted this is at fiber internet speeds, and on an array of 3 SATA SSDs. At the very least, it's possible it is exacerbated with btrfs spanning multiple disks. I learned to download my games on Linux even when I want to play them on Windows (which is less and less these days with atrocious Windows releases that run better on Linux anyway), but it's definitely reproducible and makes the driver very much unusable under these circumstances. It's pretty much perfect outside of that though, so if that could be ironed out it would be awesome.
> 
> Interesting. Does this still happen if you throttle Steam's download speeds to, say, 5 MB/s?

If this issue describes the same problem as [the issue I reported](https://github.com/maharmstone/btrfs/issues/346), then yes, it did still seem to happen with throttled speeds IIRC.

I sadly still haven't been able to reproduce it artificially/in a controlled way either.

**Comment by sebres** [comment]
> I sadly still haven't been able to reproduce it artificially/in a controlled way either.

Hmm... It's pretty reproducible with copy of large amount of data, for instance using robocopy, see https://github.com/maharmstone/btrfs/issues/438#issuecomment-991375577

**Comment by CSKnock** [comment]
Is there at least a known version that doesn't have this happening? or am I gonna have to go through all that trouble myself 😣

**Comment by alexarciniaga** [comment]
I can also confirm as of September 2023, windows 10. If I attempt to install a large steam game ~30gb. Will run smoothly for about 10 minutes writing to the drive, great speeds, 100% usage, working as expected. Then seemingly out of nowhere write and read speeds plummet to 0% usage and whole windows system freezes up until restart. 

The drive still remains in file explorer when this happens, but no longer shows any information on the drive and whole explorer will freeze up if I attempt to view properties or click on drive itself.

The strange thing is, cpu and memory are less than 30% usage before and after hang, but the driver still takes the whole system with it.

**Comment by brian6932** [comment]
I noticed a similar issue from a large torrent, and saw the following in my event viewer logs, eventually writing to this file just causes the whole system to freeze up
```
The description for Event ID 141 from source btrfs cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.

If the event originated on another computer, the display information had to be saved with the event.

The following information was included with the event: 

Drive:\torrent.!qB

The message resource is present but the message was not found in the message table
```

**Comment by smopucilowski** [comment]
I would like to debug this issue with a remote debugger on a symbols compiled driver, but I'm not familiar with windows development or windows development tools.

**Comment by brian6932** [comment]
I would like to as well, I have it built, and can build a debug build if necessary, I'm not sure where to attach a debugger for drivers though, no experience with it.

**Comment by sumariva** [comment]
Well, what I can tell is that I used the partition with zstd default(3) on to hold the bitcoin massive blockchain, 400GiB.
Since the zstd made a very good compression, about 15 to 20%, it saved me about 50GiB.
Does not seen to be related to ammount of read or write.
Because that bitcoin kept running for long hours at full speed. Read and write.

Maybe, it is leaking file descriptors.
I do not know how to activate or capture any log about the driver.

But today I got maybe a new bug. The bitcoin stopped for error, I checked the log file and it reported no more space.
But by some strange reason, the volume reports free space on windows explorer.
But by inquiring the Brtfs tab, it reports no more space left.

![btrfs-wrong-used-space](https://github.com/maharmstone/btrfs/assets/6371372/24df37e1-d6f4-4e41-a9df-3fd0bd37f3b1)

After this, I ran the scrubs thing, I supposed it be the same as chkdsk. Reported no errors.
Then, I tried the Balance with only Metadata option on but got on trouble.
As soon it begins running, a kernel fault has been detected by OS and it stopped and collect some kind of debug data.
After reboot, I stuck with a damaged volume mounted read-only.
At night will try recover from a Linux pen drive. Hope not to lose all the blocks stored on it.

![btrfs-mounted-read-only](https://github.com/maharmstone/btrfs/assets/6371372/0eaed347-4aee-4ce0-bb89-432980bf2c04)

**Comment by Masamune3210** [comment]
Just a tip? Don't store things you can't afford to lose on anything that isn't completely known to be 100% stable without a backup somewhere else

**Comment by ticpu** [comment]
Just a comment that this is still happening to me and I did learn to not do multiple operations at the same time. Copy, move, snapshots, while doing normal operations do usually result as a lock up. Respecting that, it is quite stable.

And by the way, the blockchain is the public part, it can be lost anytime, just annoying to re-download. I often use that as a benchmark for FS myself as it has its own write patterns. I usually had 4 different block chains just staying in sync and found a couple bugs in bcachefs that way.

**Comment by maharmstone** [comment]
@ticpu - I'll try that myself, then, and see what happens.

**Comment by cjvth** [comment]
Was this issue fixed in 1.9?
`Fixed hang when system under heavy load`

**Comment by maharmstone** [comment]
Yes, you're right, this is that issue.

Thank you @sumariva for the tip-off about the blockchain - it was very useful in being able to reproduce this.

**Comment by smopucilowski** [comment]
This issue persists for me to this day, and does not require heavy load to trigger.

**Comment by YoungerDryas89** [comment]
I can't believe it! this bug actually got fixed!

**Comment by maharmstone** [comment]
> I can't believe it! this bug actually got fixed!

It's okay, I'll try not to make a habit of it

**Comment by yoyossef** [comment]
I am still encountering this issue when I want to download the Metaphor Refantazio demo on Steam. I have a good connection so the download is fast, but then at around 90% it stops and hangs, and my disk becomes unaccessible.
![image](https://github.com/user-attachments/assets/c9752b84-2820-4255-87c4-deec8537f40c)

**Comment by sskras** [comment]
To everyone who still experiences the hangs / freezes / lock ups:

For how long did you wait before resetting / power cycling your machines?

I run frequenly low on resources (RAM+pagefile) on my PC, and sometimes it locks up. I was sure it's not related to WinBTRFS driver. Several days ago it froze again after painting the screen black. Instead of resetting the laptop I left it in that state for 15 minutes or so. And you know what? When I came back, it was unfrozen, actually responsible!

Also this reminded me of some of the freezing related issues of WinBTRFS. So I thought would share the idea: 

maybe waiting for a bit longer would give some additional insights into your issue.

**Comment by lesderid** [comment]
> To everyone who still experiences the hangs / freezes / lock ups:
> 
> For how long did you wait before resetting / power cycling your machines?

At one time, multiple days. Most recently, overnight.

**Comment by yoyossef** [comment]
Following my previous comment, I have installed a similarly sized game (Metro Exodus Enhanced Edition) and I didn't encounter any issues.

I wonder if it is due to Metaphor Refantazio packaging their game weirdly. The game mostly relies on one ".cpk" file that weighs [53.68 GiB](https://steamdb.info/depot/3130331/).

So when Steam decompresses such a big file, WinBTRFS starts failing somehow? Is this a possible scenario?

BTW the prologue is completely free on Steam if anyone want to hop in and say if they reproduce the issue: https://store.steampowered.com/app/3130330/

> To everyone who still experiences the hangs / freezes / lock ups:
>
> For how long did you wait before resetting / power cycling your machines?

Hours, didn't change, also it'd block shutting the PC down with an infinite "Powering Off" screen from Windows.

**Comment by rautamiekka** [comment]
> Hours, didn't change, also it'd block shutting the PC down with an infinite "Powering Off" screen from Windows.

Using the commands `poweroff /s /t 0 /f` for shutdown and `poweroff /r /t 0 /f` for reboot might work, since the OS should forcibly kill whatever is blocking it (not guaranteed), but I suspect there's a good risk of corruption/loss as a result.

**Comment by ticpu** [comment]
> Using the commands `poweroff /s /t 0 /f` for shutdown and `poweroff /r /t 0 /f` for reboot might work

That doesn't work on stuck kernel threads.

I've been power cycling, usually after 15 minutes. Doesn't cause any more data corruption, if it is locked up, pending writes just stack up and any changes just don't appear after reboot.

**Comment by rautamiekka** [comment]
> > Using the commands `poweroff /s /t 0 /f` for shutdown and `poweroff /r /t 0 /f` for reboot might work
> 
> That doesn't work on stuck kernel threads.
> 
> I've been power cycling, usually after 15 minutes. Doesn't cause any more data corruption, if it is locked up, pending writes just stack up and any changes just don't appear after reboot.

Good stuff to know.

**Comment by jieran233** [comment]
I can conform the `1.9` version still occurs this bug, at least they have same behavior.

**Comment by Wemmy0** [comment]
Yeah I keep having issues where Steam will lock up, then eventually explorer will slowly start to crash eventually causing the system to be unusable but still "running". The only way I've been able to get it to power down is by unplugging the power cable.

**Comment by jarrard** [comment]
I've read a few reports of this being MORE of a issue when compression is enabled on the drive?

**Comment by namquang93** [comment]
Having the same issue here, especially when copying large files to btrfs partition.

**Comment by popcar2** [comment]
Still happens on the newest update... Deleting a larger file (1.5gb+) is almost guaranteed to disconnect the drive and maybe even corrupt the file.

**Comment by Shroopy** [comment]
Happens consistently for me, had to stop using winbtrfs, this issue should still be open

**Comment by jarrard** [comment]
Yeah winbtrfs still needs a bit of work. It's like 90% of the way to the finish line of being reliably used.
